app/server/resourceTypes/mapperResource.ts

```diff
@@ -0,0 +1,126 @@
+import {Resource} from 'rest-io';
+import {Response, Request} from 'express';
+import {Document} from 'mongoose';
+import request = require('request');
+
+interface IMap {
+  place: string
+  source: string,
+  destination: string,
+  value: string
+}
+
+interface IMapper {
+  maps: [IMap]
+}
+
+interface IRequest {
+  url: string,
+  data: any,
+  headers: any
+}
+
+class MapperResource extends Resource {
+
+
+  // will extract a value of an object using a map
+  // a map can be a:
+  //   * 'foo' - gets the property `foo` from the root of the provided object
+  //   * 'foo.bar.baz' - you can chain property names to get deeper into the object
+  //   * 'foo[0].bar' - goes into array `foo` index 0 and gets the object `bar`
+  //   * 'foo[0][1][0].baz' - you can also go into nested arrays
+  singleMap(obj: Object, map: String) {
+    map = map.replace(/\]/g, '').replace(/\[/g, '.');
+
+    var parts = map.split('.');
+    var currentPointer = obj;
+    for (var i = 0; i < parts.length; i++) {
```

Comment:
Why not 

    parts.forEach((part: string) => currentPointer = currentPointer[part]);